tools/src/main/java/org/apache/kafka/trogdor/workload/ConsumeBenchWorker.java

```diff
@@ -202,6 +257,9 @@ public Void call() throws Exception {
                         latencyHistogram.add(elapsedBatchMs);
                         messageSizeHistogram.add(messageBytes);
                         bytesConsumed += messageBytes;
+                        if (messagesConsumed >= maxMessages)
```

Comment:
Why is this needed?  The loop condition is the same.

Reply:
The default `max.poll.records` is 500. This would result in the consumer always consuming 500 at a minimum (provided the topic had the records) and not respect the cases where `maxMessages` was less than that